src/core/border_router/routing_manager.cpp

```diff
@@ -1940,6 +1913,16 @@ Error RoutingManager::RxRaTracker::Iterator::AdvanceToNextEntry(void)
 //---------------------------------------------------------------------------------------------------------------------
 // RxRaTracker::Router
 
+bool RoutingManager::RxRaTracker::Router::ShouldCheckReachability(void) const
+{
+    // Perform reachability check (send NS probes) only if the router:
```

Comment:
Does this method actually perform the reachability check as indicated by this comment? If so, it surely does this unconditionally since `isReachable is on the left side of the expression`)

Reply:
It determines whether or not a router's reachability should be checked.